python/cudf/io/csv.py

```diff
@@ -264,6 +275,10 @@ def read_csv_strings(filepath, lineterminator='\n',
     csv_reader.num_cols = len(names)
     csv_reader.skiprows = skiprows
     csv_reader.skipfooter = skipfooter
+    csv_reader.decimal = decimal.encode()
+    csv_reader.thousands = ffi.NULL
+    if thousands:
+        csv_reader.thousands = ffi.new('char*', thousands.encode())
```

Comment:
You can use csv_reader.thousands = _wrap_string(thousands) since that function basically does all these lines.

Aside from that, the thousands parameter can only be a 1-length string in PANDAS. Can we change the C++ argument to be a char instead of a char* ? That will match decimal and other single-character arguments and we can simplify it by avoiding a nullptr check.

Reply:
Thousands parameter is not a string, it is a pointer to a single character. This is why I used 
ffi.new('char*' instead of ffi.new('char[]'
the first option allocates only a single character.
Because of this difference, I cannot use the _wrap_string function here.
Also, AFAIK the current implementation is consistent with pandas, as it can only be a single character, and is optional.

Reply:
We deviate from PANDAS for all other similar parameters that are "1-length strings" since there isn't a really a character type as in C/C++. In the C/C++ interface, we can simply use a character. Why use up more space to store a pointer when you can store a character in the struct?

At the least, there should not be any shouldn't be inconsistency with the other parameters.

Reply:
Will address this is a separate PR, to keep the scope of this PR to read_csv_strings().